client/my-sites/patterns/components/pattern-library/index.tsx

```diff
@@ -103,6 +104,16 @@ export const PatternLibrary = ( {
 		},
 	} );
 
+	let patternPermalinkName;
+	if ( patternPermalinkId && patterns.length ) {
+		for ( const pattern of patterns ) {
+			if ( pattern.ID === patternPermalinkId ) {
+				patternPermalinkName = pattern.name;
+				break;
+			}
+		}
+	}
```

Comment:
I would suggest relying on `isFetchingPatterns` to determine the availability of the `patterns` array, and we can eliminate indentation as follows:

```suggestion
	if ( ! isFetchingPatterns && patternPermalinkId ) {
		const pattern = patterns.find( ( pattern ) => pattern.ID === patternPermalinkId );
		patternPermalinkName = pattern ? pattern.name : undefined;
	}
```

Reply:
I think we could even simplify the two last lines from Angela's suggestion like so:

```javascript
patterns.find( pattern => pattern.ID === patternPermalinkId )?.name;
```